sql/parser: TestParseDatadriven failed

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7600198?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7600198?buildTab=artifacts#/) on release-22.2 @ [022d19ab6c1c6f6dd477d5120b4c47a46f744238](https://github.com/cockroachdb/cockroach/commits/022d19ab6c1c6f6dd477d5120b4c47a46f744238):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/709/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/709/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/709/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21640

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7600194?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7600194?buildTab=artifacts#/) on release-22.2 @ [022d19ab6c1c6f6dd477d5120b4c47a46f744238](https://github.com/cockroachdb/cockroach/commits/022d19ab6c1c6f6dd477d5120b4c47a46f744238):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/956/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/956/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/956/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7600205?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7600205?buildTab=artifacts#/) on release-22.2 @ [022d19ab6c1c6f6dd477d5120b4c47a46f744238](https://github.com/cockroachdb/cockroach/commits/022d19ab6c1c6f6dd477d5120b4c47a46f744238):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1624/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1624/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1624/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7611465?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7611465?buildTab=artifacts#/) on release-22.2 @ [bf7c1cf83275be4645fc77581be2cb9d5e8502c4](https://github.com/cockroachdb/cockroach/commits/bf7c1cf83275be4645fc77581be2cb9d5e8502c4):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1234/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1234/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1234/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7611472?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7611472?buildTab=artifacts#/) on release-22.2 @ [bf7c1cf83275be4645fc77581be2cb9d5e8502c4](https://github.com/cockroachdb/cockroach/commits/bf7c1cf83275be4645fc77581be2cb9d5e8502c4):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/788/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/788/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/788/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7611469?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7611469?buildTab=artifacts#/) on release-22.2 @ [bf7c1cf83275be4645fc77581be2cb9d5e8502c4](https://github.com/cockroachdb/cockroach/commits/bf7c1cf83275be4645fc77581be2cb9d5e8502c4):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2122/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2122/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2122/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7623867?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7623867?buildTab=artifacts#/) on release-22.2 @ [bf7c1cf83275be4645fc77581be2cb9d5e8502c4](https://github.com/cockroachdb/cockroach/commits/bf7c1cf83275be4645fc77581be2cb9d5e8502c4):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1437/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1437/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1437/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7623863?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7623863?buildTab=artifacts#/) on release-22.2 @ [bf7c1cf83275be4645fc77581be2cb9d5e8502c4](https://github.com/cockroachdb/cockroach/commits/bf7c1cf83275be4645fc77581be2cb9d5e8502c4):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1488/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1488/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1488/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/parser.TestParseDatadriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7623870?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7623870?buildTab=artifacts#/) on release-22.2 @ [bf7c1cf83275be4645fc77581be2cb9d5e8502c4](https://github.com/cockroachdb/cockroach/commits/bf7c1cf83275be4645fc77581be2cb9d5e8502c4):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1113/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:216:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "inout": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(INOUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                     ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1113/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:237:
        error [0 args]
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        ----
        at or near "out": syntax error: unimplemented: this syntax
        DETAIL: source SQL:
        CREATE OR REPLACE FUNCTION f(IN OUT a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                        ^
        HINT: You have attempted to use a feature that is not yet implemented.
        
        Please check the public issue tracker to check whether this problem is
        already tracked. If you cannot find it there, please report the error
        with details by creating a new issue.
        
        If you would rather not post publicly, please contact us directly
        using the support form.
        
        We appreciate your feedback.
    parse_test.go:39: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1113/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/parser/parser_test_/parser_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/parser/testdata/create_function:258: CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
        output didn't match expected:
        @@ -1,7 +1,7 @@
         at or near "variadic": syntax error: unimplemented: this syntax
         DETAIL: source SQL:
         CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                                      ^
         HINT: You have attempted to use a feature that is not yet implemented.
        -See: https://go.crdb.dev/issue-v/88947/v22.2
        +See: https://go.crdb.dev/issue-v/88947/dev
         
    --- FAIL: TestParseDatadriven/create_function (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92053 sql/parser: TestParseDatadriven failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseDatadriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/parser/parse_test.go
+++ b/pkg/sql/parser/parse_test.go
@@ -15,6 +15,7 @@ import (
 	"fmt"
 	"go/constant"
 	"reflect"
+	"regexp"
 	"strings"
 	"testing"
 
@@ -31,6 +32,8 @@ import (
 	"github.com/cockroachdb/errors"
 	"github.com/stretchr/testify/assert"
 )
+
+var issueLinkRE = regexp.MustCompile("https://go.crdb.dev/issue-v/([0-9]+)/.*")
 
 // TestParseDataDriven verifies that we can parse the supplied SQL and regenerate the SQL
 // string from the syntax tree.
@@ -98,6 +101,7 @@ func TestParseDatadriven(t *testing.T) {
 				if pgerr.Hint != "" {
 					msg += "\nHINT: " + pgerr.Hint
 				}
+				msg = issueLinkRE.ReplaceAllString(msg, "https://go.crdb.dev/issue-v/$1/")
 				return msg
 			}
 			d.Fatalf(t, "unsupported command: %s", d.Cmd)

--- a/pkg/sql/parser/testdata/create_function
+++ b/pkg/sql/parser/testdata/create_function
@@ -263,7 +263,7 @@ DETAIL: source SQL:
 CREATE OR REPLACE FUNCTION f(VARIADIC a int = 7) RETURNS INT AS 'SELECT 1' LANGUAGE SQL
                              ^
 HINT: You have attempted to use a feature that is not yet implemented.
-See: https://go.crdb.dev/issue-v/88947/v23.1
+See: https://go.crdb.dev/issue-v/88947/
 
 error
 CREATE OR REPLACE FUNCTION f(a int = 7) RETURNS INT TRANSFORM AS 'SELECT 1' LANGUAGE SQL

